diff --git a/dist/index.mjs b/dist/index.mjs
index f74a06dc3087751e10f97a0cba84ccfb77ed80d1..806a10b66b4e557c6d7df672f8a6ce1c3fda579e 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1292,7 +1292,12 @@ var GoogleGenerativeAILanguageModel = class {
                   controller.enqueue({
                     type: "file",
                     mediaType: part.inlineData.mimeType,
-                    data: part.inlineData.data
+                    data: part.inlineData.data,
+                    providerMetadata: part.thoughtSignature ? {
+                      [providerOptionsName]: {
+                        thoughtSignature: part.thoughtSignature
+                      }
+                    } : void 0
                   });
                 }
               }
